pkg/resource/logging-config/promtail-logging-config.go

```diff
@@ -1,230 +1,42 @@
 package loggingconfig
 
 import (
-	"github.com/pkg/errors"
-	"sigs.k8s.io/yaml"
+	"bytes"
+	"text/template"
+
+	_ "embed"
+
+	"github.com/Masterminds/sprig/v3"
 
 	"github.com/giantswarm/logging-operator/pkg/common"
 	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
 )
 
-// /// Promtail values config structure
-type values struct {
-	Promtail promtail `yaml:"promtail" json:"promtail"`
-}
-
-type promtail struct {
-	ExtraArgs         []string                   `yaml:"extraArgs" json:"extraArgs"`
-	ExtraEnv          []promtailExtraEnv         `yaml:"extraEnv" json:"extraEnv"`
-	Config            promtailConfig             `yaml:"config" json:"config"`
-	ExtraVolumes      []promtailExtraVolume      `yaml:"extraVolumes" json:"extraVolumes"`
-	ExtraVolumeMounts []promtailExtraVolumeMount `yaml:"extraVolumeMounts" json:"extraVolumeMounts"`
-}
-
-type promtailExtraEnvValuefrom struct {
-	FieldRef promtailExtraEnvFieldref `yaml:"fieldRef" json:"fieldRef"`
-}
-
-type promtailExtraEnvFieldref struct {
-	FieldPath string `yaml:"fieldPath" json:"fieldPath"`
-}
-
-type promtailExtraEnv struct {
-	Name      string                    `yaml:"name" json:"name"`
-	ValueFrom promtailExtraEnvValuefrom `yaml:"valueFrom" json:"valueFrom"`
-}
-
-type promtailConfigSnippets struct {
-	ExtraScrapeConfigs  string               `yaml:"extraScrapeConfigs" json:"extraScrapeConfigs"`
-	ExtraRelabelConfigs []extraRelabelConfig `yaml:"extraRelabelConfigs" json:"extraRelabelConfigs"`
-	AddScrapeJobLabel   bool                 `yaml:"addScrapeJobLabel" json:"addScrapeJobLabel"`
-}
-
-type promtailConfig struct {
-	Snippets promtailConfigSnippets `yaml:"snippets" json:"snippets"`
-}
-
-type promtailExtraVolume struct {
-	Name     string                      `yaml:"name" json:"name"`
-	HostPath promtailExtraVolumeHostpath `yaml:"hostPath" json:"hostPath"`
-}
-
-type promtailExtraVolumeHostpath struct {
-	Path string `yaml:"path" json:"path"`
-}
-
-type promtailExtraVolumeMount struct {
-	Name      string `yaml:"name" json:"name"`
-	MountPath string `yaml:"mountPath" json:"mountPath"`
-	ReadOnly  bool   `yaml:"readOnly" json:"readOnly"`
-}
+var (
```

Comment:
Replace go struct with template as pipeline stages would be of the type []map[string]interface{} but that cannot be marshalled so going with a template is definitely easier and it also reduces the code